src/patches.js

```diff
@@ -125,15 +141,7 @@ export function applyPatches(draft, patches) {
                     base[key] = patch.value
                     break
                 case "remove":
-                    if (Array.isArray(base)) {
```

Comment:
I'm fine by just explicitly forbidding it (I mean, it is pretty weird to call `delete` on an array, as `splice` is the idiomatic way to remove entries in the middle). That als simplifies the rest of the implementation

Reply:
In both proxy and es5 mode, when delete an array item wont't cause any error, unless calling applyPatches with the patches. I think it's better to use the patches to replay the actions

Reply:
for example
```
   let anArray = [1,2,3];
   let changes = [];
   produce(a, d => {
       delete d[1];
   }, (p, i) => {
       changes = [...p];
   })
  
  let arr1 = applyPatches([1, 2], changes)
  console.log(arr1)  // [1]

  let arr2 = applyPatches([1, 2, 3, 4], changes)
  console.log(arr2) //  throw "Remove can only remove the last key of an array"
```
the same patches apply on some data will lead to  different result. It's so confused for me.

Reply:
I would simplify the entire solution to have the proxy trap throw on delete if the target is an array. `delete` should not be used on arrays in producers period. It will keep the implementation of immer  faster as well, compared to the current changes in the PR. If it was super easy it would be fine to support, but there is no real value in supporting it imho, so no reason to jump through hoops to get it working.

Reply:
Ok, that's may be a good idea.